obc/drivers/vn100/vn100_binary.c

```
@@ -0,0 +1,80 @@
+  const char checksum[] = ",01,0528*XX\r\n";
+
+  snprintf(outputRateString, sizeof(outputRateString), "%d", outputRate);
+
+  size_t headerLength = strlen(header);
+  size_t outputRateLength = strlen(outputRateString);
+  size_t checksumLength = strlen(checksum);
+
+  unsigned char buf[MAX_SEND_SIZE];
+
+  // Begin appending the command
+  memcpy(buf, header, headerLength);
+  memcpy(buf + headerLength, outputRateString, outputRateLength);
+  memcpy(buf + headerLength + outputRateLength, checksum, checksumLength);
+
+  obc_error_code_t errCode;
+  RETURN_IF_ERROR_CODE(sciSendBytes(buf, sizeof(buf), portMAX_DELAY, UART_VN100_REG));
+  return OBC_ERR_CODE_SUCCESS;
+}
+
+obc_error_code_t stopBinaryOutputs(void) {
+  // unsigned char buf[] = "$VNWRG,75,0,80,01,0528*XX\r\n";
+
+  uint8_t outputRate = 800 / DEFAULT_BINARY_OUTPUT_RATE_HZ;  // See section
+
+  const char header[] = "$VNWRG,75,0,";
+  char outputRateString[MAX_OUTPUT_RATE_SIZE];  // Sufficient space for an int
+  const char checksum[] = ",01,0528*XX\r\n";
+
+  snprintf(outputRateString, sizeof(outputRateString), "%d", outputRate);
+
+  size_t headerLength = strlen(header);
+  size_t outputRateLength = strlen(outputRateString);
+  size_t checksumLength = strlen(checksum);
+
+  unsigned char buf[MAX_SEND_SIZE];
+
+  // Begin appending the command
+  memcpy(buf, header, headerLength);
+  memcpy(buf + headerLength, outputRateString, outputRateLength);
+  memcpy(buf + headerLength + outputRateLength, checksum, checksumLength);
+
+  obc_error_code_t errCode;
+  RETURN_IF_ERROR_CODE(sciSendBytes(buf, sizeof(buf), portMAX_DELAY, UART_VN100_REG));
+  return OBC_ERR_CODE_SUCCESS;
+}
+
+obc_error_code_t readBinaryOutputs(vn100_binary_packet_t* parsedPacket) {
+  unsigned char buf[BINARY_PACKET_SIZE] = {'\0'};
+  obc_error_code_t errCode;
+  RETURN_IF_ERROR_CODE(sciReadBytes(buf, BINARY_PACKET_SIZE, portMAX_DELAY, pdMS_TO_TICKS(1000), UART_VN100_REG));
```

1 second seems like a long time? If it's outputting at 10Hz, then a safe value here would be 2x the output rate. Otherwise, something's wrong and we should return an error code.